### ISSUE TITLE
Encoding option does not work

### ISSUE THREAD

**Issue by omgold** [wkhtmltopdf/wkhtmltopdf#4492]
**wkhtmltopdf version(s) affected**: 0.12.5-20-ge3fa5a9

**OS information**

Distro: Arch Linux
qt5-webkit version: 

**Description**

The --encoding option is ignored (on a build with default options). For (some?) utf-8 input files without charset specification non-ascii characters are garbage.

**How to reproduce**

Invocation like this:

```
wkhtmltopdf --encoding utf-8 infile.html outfile.pdf
```

On the this input.html file:

```
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
<html xmlns="http://www.w3.org/1999/xhtml"><body>
Umlauts in UTF-8 encoding: ÄÖÜäöü
</body></html>
```

**Expected behavior**

Non-Ascii UTF-8 encoded characters are rendered correctly.

**Possible Solution**  

The code to apply the encoding setting seems to be disabled( depending on the define  EXTENSIVE_WKHTMLTOPDF_QT_HACK). Enabling it unconditionally fixes the issue for me:

```
--- src/lib/converter.cc.orig   2019-10-04 10:50:49.335597221 +0200
+++ src/lib/converter.cc        2019-10-04 10:50:41.878374951 +0200
@@ -36,9 +36,9 @@
 
 
 void ConverterPrivate::updateWebSettings(QWebSettings * ws, const settings::Web & s) const {
-#ifdef  __EXTENSIVE_WKHTMLTOPDF_QT_HACK__
        if (!s.defaultEncoding.isEmpty())
                ws->setDefaultTextEncoding(s.defaultEncoding);
+#ifdef  __EXTENSIVE_WKHTMLTOPDF_QT_HACK__
        if (!s.enableIntelligentShrinking) {
                ws->setPrintingMaximumShrinkFactor(1.0);
                ws->setPrintingMinimumShrinkFactor(1.0);

```

**Comment by tkuisma** [comment]
Hi, I'm struggling with this same problem, if html file has non-ascii characters.

The resulting pdf is always ASCII-8BIT so when used, it will result encoding error
Encoding::UndefinedConversionError "\xFE" from ASCII-8BIT to UTF-8

Noticed similar issue about the problem
https://github.com/wkhtmltopdf/wkhtmltopdf/issues/3233

**Comment by ashkulz** [comment]
Fixed by #4612
